Lib/test/test_concurrent_futures.py

```diff
@@ -1018,6 +1018,14 @@ def test_idle_process_reuse_multiple(self):
         self.assertLessEqual(len(executor._processes), 2)
         executor.shutdown()
 
+    def test_maxtasksperchild(self):
+        executor = self.executor_type(1, maxtasksperchild=1)
+        executor.submit(init, "init").result()
+        status = executor.submit(get_init_status).result()
+        self.assertEqual(status, "uninitialized")
+        executor.shutdown()
```

Comment:
I would expect the test to actually check that exactly `max_tasks_per_child` tasks are spawned in a given worker process instance.

Perhaps the existing tests for `multiprocessing.Pool` can be an inspiration.

Reply:
Ok, I'll take a look at the tests and see if I can come up with something better.

Reply:
Alright, while I was working on additional tests, I found a new bug, so I'm trying to fix that before I push up the changes. Basically, if the executor shuts down before the futures are complete, an assertion error occurs and then a thread hangs. Will re-push once this is fixed.

Reply:
You said you'be working on additional tests, but are they part of the PR now?

Reply:
Yes, there are now two tests `test_max_tasks_per_child` and the new `test_max_tasks_early_shutdown`.